src/lib/lore.sh

```diff
@@ -411,112 +411,77 @@ function get_patches_from_mailing_list()
   done
 }
 
-function title_to_path()
-{
-  local title="$1"
-  local file_name
-
-  # Replace space in favor of _
-  file_name="${title// /_}"
-
-  # Replace .
-  file_name="${file_name//./_}"
-
-  # Replace special character
-  file_name="${file_name//[&*+%!:,]/}"
-
-  # Replace /
-  file_name="${file_name//[\/\\]/-}"
-
-  printf '%s' "$file_name"
-}
-
-function convert_title_to_patch_name()
-{
-  local title="$1"
-  local index="$2"
-  local file_name
-
-  file_name=$(title_to_path "$title")
-
-  if [[ -n "$index" ]]; then
-    printf '%04d-%s\n' "$index" "${file_name}.mbox"
-    return
-  fi
-
-  printf '%s\n' "${file_name}.mbox"
-}
-
-function url_update_patch_number()
-{
-  local url="$1"
-  local new_number="$2"
-
-  url="${url/-[0-9]*-/-$link_ref-}"
-
-  printf '%s' "$url"
-}
-
+# This function downloads a patch series in a .mbx format to a given directory
+# using a series URL. The series URL should be the concatenation of the lore URL, the
+# target mailing list, and the message ID. Below is an example of such a URL:
+#   https://lore.kernel.org/some-list/2367462342.4535535-1-email@email.com/
+# The output filename is '<message ID>.mbx'. This function uses the `b4` tool underneath
+# to delegate the process of fetching and downloading the series. The file generated is
+# ready to be applied into a git tree, containing just the patches in the right order,
+# without the cover letter.
+#
+# Return:
+# Return 0 if the thread was successfully downloaded, 22 if the series URL or the output
+# directory passed as arguments is empty and the error code of `b4` in case it fails.
 function download_series()
 {
-  local total_patches="$1"
-  local first_message_id="$2"
-  local save_to="$3"
-  local title="$4"
-  local total=0
-  local link_ref=0
+  local series_url="$1"
+  local save_to="$2"
+  local flag=${3:-'SILENT'}
   local url
-  local patch_file_name
+  local series_filename
+  local cmd
+  local ret
 
-  mkdir -p "$save_to"
+  # Safety checking
+  if [[ -z "$series_url" || -z "$save_to" ]]; then
+    return 22 # EINVAL
+  fi
 
-  url=$(replace_http_by_https "$first_message_id")
+  # Create the output directory if it doesn't exists
+  cmd_manager "$flag" "mkdir --parents '${save_to}'"
+  ret="$?"
+  [[ "$ret" != 0 ]] && return "$ret"
```

Comment:
Nice. I didn't return a message to the user, because any output to the stdin in this feature is overwritten by dialog screens, resulting in a flickering message. However, your message made me think that the view can capture this message and display it via dialog. This makes the error thrown when `download_series` fails more fine-grained.